### ISSUE TITLE
conflict between Style/ArgumentsForwarding and new block ambiguity rules in Ruby 3.3

### ISSUE THREAD

**Issue by Roguelazer** [rubocop/rubocop#12608]
On Ruby 3.3, it's illegal to use anonymous arguments inside a block; the following code is legal on Ruby 3.2, but raises "anonymous keyword rest parameter is also used within block (SyntaxError)" on Ruby 3.3:

```ruby
def bar(a:)
  puts a
end

def foo(**)
  -> { bar(**) }
end

foo(a: :baz).call
```

Unfortunately, even with TargetRubyVersion set to 3.3, if Style/ArgumentsForwarding is enabled, rubocop enforces that the `**` be anonymous.

Note that this may or may not be changed for this particular case in Ruby 3.3.1 (see https://github.com/ruby/ruby/commit/596db9c1f486d6609a4e97d82c8c71b54609fb6f), but it does appear to still be illegal for some potentially-ambiguous cases.

--------

## Expected behavior

I'm not sure; it'd be cool to do some semantic analysis to determine whether a block is constructed in the lexical scope of the anonymous arguments and only rewrite it if it's _not_ used inside a block.

## Actual behavior

Rubocop insists on removing the name from the anonymous argument and makes the file have a syntax error.

## Steps to reproduce the problem

Create the following file as `test.rb`:

```ruby
# frozen_string_literal: true

def bar(arg:)
  puts arg
end

def foo(**kwargs)
  -> { bar(**kwargs) }
end

foo(arg: :baz).call
```

Create the following file as `rubocop.yml`:

```yaml
AllCops:
  TargetRubyVersion: 3.3
Style/ArgumentsForwarding:
  Enabled: true
```

Note that `rubocop --fix test.rb` will attempt to remove the name from `kwargs`, which then makes the file have a syntax error on Ruby 3.3.

## RuboCop version
```
$ ~/.gem/ruby/3.3.0/bin/rubocop -V
1.59.0 (using Parser 3.3.0.2, rubocop-ast 1.30.0, running on ruby 3.3.0) [arm64-darwin23]
```

**Comment by koic** [comment]
This issue is a dup with #12571. Thank you for your feedback.
